### ISSUE TITLE
Validation performance inconsistent during critic warmup

### ISSUE THREAD

**Issue by Squidwarder** [volcengine/verl#1209]
Hello, I'm currently experiencing an issue with PPO training where during critic warmup (thus **actor is frozen**), the accuracy on validation test varies.

<img width="1074" alt="Image" src="https://github.com/user-attachments/assets/31319a23-94ce-4e45-ac9a-e27a6d29c728" />

<img width="647" alt="Image" src="https://github.com/user-attachments/assets/c25dc091-29d0-4f4e-8bff-6b7ca86c4ec2" />

Like the image shown here I've set my vLLM params for testing to be `temperature=0`, so in theory the accuracy should stay unchanged because the actor model is frozen during critic warmup (step=20). Instead, I'm observing changes in validation accuracy (my validation set is 125 data). Is this an issue with vLLM or something else?

My vLLM (v0 engine) config is as follows: 
```
VLLM_ATTENTION_BACKEND=XFORMERS
# sampling parameters for validation
top_k: -1 # 0 for hf rollout, -1 for vllm rollout
top_p: 1.0
temperature: 0
n: 1
do_sample: False # default eager for validation
```

**Comment by eric-haibin-lin** [comment]
can u reproduce this issue with [main_generation](https://github.com/volcengine/verl/blob/main/verl/trainer/main_generation.py) and observe different outputs? If so that sounds like a non-deterministic behavior from vllm.

**Comment by Squidwarder** [comment]
Thank you so much for responding. By using [main_generation](https://github.com/volcengine/verl/blob/main/verl/trainer/main_generation.py), do you mean that I should run it multiple times with fixed model (as well as generation config) and input file, then compare the parquet files the script produces?

**Comment by Squidwarder** [comment]
On further investigation, it appears that the problem was due to mismatch in vLLM version for all nodes involved (two uses 0.8.2, and two uses 0.8.1). Once they are aligned, the validation accuracy becomes more closely matched (there are still a couple of small fluctuations here and there, but I assume that's just vLLM behavior and can't be helped).

**Comment by Prathyusha-Guduru** [comment]
How many warmup steps are recommended?
